I haven't seen this before:
```
2023-07-25T07:32:11.1026456Z === RUN   TestHealObjectCorruptedPools
2023-07-25T07:32:11.1027294Z     erasure-healing_test.go:1070: Expected xl.meta file to be not present, but succeeeded
...
2023-07-25T07:32:11.1040672Z --- FAIL: TestHealObjectCorruptedPools (0.63s)
```

--- a/cmd/erasure-healing_test.go
+++ b/cmd/erasure-healing_test.go
@@ -1065,8 +1065,8 @@ func TestHealObjectCorruptedPools(t *testing.T) {
 	}
 
 	for i := 0; i < (nfi.Erasure.DataBlocks + nfi.Erasure.ParityBlocks); i++ {
-		_, err = erasureDisks[i].StatInfoFile(context.Background(), bucket, pathJoin(object, xlStorageFormatFile), false)
-		if err == nil {
+		stats, _ := erasureDisks[i].StatInfoFile(context.Background(), bucket, pathJoin(object, xlStorageFormatFile), false)
+		if len(stats) != 0 {
 			t.Errorf("Expected xl.meta file to be not present, but succeeeded")
 		}
 	}

--- a/cmd/xl-storage.go
+++ b/cmd/xl-storage.go
@@ -1443,8 +1443,8 @@ func (s *xlStorage) ReadVersion(ctx context.Context, volume, path, versionID str
 			// Check the data path if there is a part with data.
 			partPath := fmt.Sprintf("part.%d", fi.Parts[0].Number)
 			dataPath := pathJoin(path, fi.DataDir, partPath)
-			_, err = s.StatInfoFile(ctx, volume, dataPath, false)
-			if err != nil {
+			_, lerr := Lstat(pathJoin(volumeDir, dataPath))
+			if lerr != nil {
 				// Set the inline header, our inlined data is fine.
 				fi.SetInlineData()
 				return fi, nil
@@ -2723,7 +2723,11 @@ func (s *xlStorage) StatInfoFile(ctx context.Context, volume, path string, glob 
 		}
 		st, _ := Lstat(filePath)
 		if st == nil {
-			continue
+			// Stat a volume entry.
+			if verr := Access(volumeDir); verr != nil {
+				return stat, convertAccessError(verr, errVolumeAccessDenied)
+			}
+			return stat, errPathNotFound
 		}
 		name, err := filepath.Rel(volumeDir, filePath)
 		if err != nil {

